text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,495 @@
+- **Dropped**: Considered failure, an item of telemetry was eliminated (i.e., export never attempted)
+
+The second equation:
+
+```
+Outgoing(Segment) == Discarded(Segment) + Dropped(Segment) + Exported(Segment)
+```
+
+Note that the incoming categories match the origin attribute, while the 
+outgoing categories look like a pared down version of the outcome attribute. 
+However, the exported category here encompasses all outcomes where export 
+is attempted.  The `exported` outcome is only when export is successful.
+
+The third equation states that the sum of all incoming items should equal the 
+sum of all outgoing items.
+
+```
+Incoming(Segment) == Outgoing(Segment)
+```
+
+#### Invariant checking
+
+In order to use these equations exactly as written, it is important to
+recognize that incoming and outgoing accounting is independent and
+that these signals can lead or lag each order, depending on pipeline
+configuration.
+
+The simplest way to check the invariants, therefore, is to "drain" the
+pipeline so that it is empty, meaning all items have been accounted
+for. For the OpenTelemetry Collector, therefore, it should be
+possible to verify this and warn about improper accounting during
+shutdown. After a pipeline has been drained and the individual
+components shut down, we expect the incoming and outgoing instrument
+values to match exactly.
+
+### Measuring Pipeline Loss Between Segments
+
+Considering an Exporter and Receiver pair connecting two OpenTelemetry
+Collector pipeline segments, we expect ideally the number exported to
+match the number received.
+
+```
+Exported(Preceder) == Received(Follower)
+```
+
+Where Exported and Received counts are defined by the first and second
+equations above, noting that Exported excludes Dropped or Discarded
+outcomes.
+
+When an exporter retries the same item on failures, the equality changes:
+
+```
+Exported(Preceder) <= Received(Follower)
+```
+
+However, some outcomes in the preceder don't indicate that the follow ever 
+received (and therefore couldn't count) the item. This is in the case where a 
+timeout has occurred on the exporter side before any response from the 
+follower. In this case, the equality shifts the other way:
+
+```
+Exported(Preceder) >= Received(Follower)
+```
+
+We can describe exported items that have definitely received a response from 
+the follower by removing `retryable` and `timeout`:
+
+```
+Exported_Responded(Segment) == Exported(Segment) - Retryable(Segment) - Timeout(Segment)
+```
+
+In a real pipeline, therefore, we expect:
+
+```
+Exported_Responded(Preceder) <= Received(Follower)
+```
+
+When an observability platform detects an out-of-range value, there is
+likely to be a defective pipeline or missing instrumentation.
+
+### Measuring Total Pipeline Loss
+
+*WIP: Figure this out. This is a bit subjective. What does an end user expect 
+when calculating total items dropped in failure?*
```

Comment:
To me, the idea of "total" signifies starting from the beginning of the pipeline at the SDKs, meaning to use a sum of all the SDK-inserted items and compare against some point later in the pipeline to measure how many are original items are lost somehow. This could mean looking at a gateway collector's exporter counts and comparing to the SDK-inserted counts, for example.